tests/test_util.py

```diff
@@ -115,6 +115,36 @@ def test_image_name_format():
         assert image_name.to_str() == expected
 
 
+@pytest.mark.parametrize(('repo', 'organization', 'enclosed_repo'), (
+    ('fedora', 'spam', 'spam/fedora'),
+    ('spam/fedora', 'spam', 'spam/fedora'),
+    ('spam/fedora', 'maps', 'maps/spam-fedora'),
+))
+@pytest.mark.parametrize('registry', (
+    'example.registry.com',
+    'example.registry.com:8888',
+    None,
+))
+@pytest.mark.parametrize('tag', ('bacon', None))
+def test_image_name_enclose(repo, organization, enclosed_repo, registry, tag):
+    reference = repo
+    if tag:
+        reference = '{}:{}'.format(repo, tag)
+    if registry:
+        reference = '{}/{}'.format(registry, reference)
+
+    image_name = ImageName.parse(reference)
+    assert image_name.get_repo() == repo
+    assert image_name.registry == registry
+    assert image_name.tag == tag
```

Comment:
This line needs changing once https://github.com/projectatomic/atomic-reactor/pull/1062 is merged:
```python
assert image_name.tag == (tag or 'latest')
```

Reply:
Updated.